backend/cmd/api/internal/model/questions.go

```diff
@@ -63,69 +50,39 @@ func (q *Question) Save(ctx context.Context) error {
 
 // FindQuestions returns questions without joins, it is used by admins for management
 func FindQuestions(ctx context.Context) ([]*Question, error) {
-	sql, boundArgs, _ := sqlBuilder.
+	sqlb := stmntBuilder.
 		Select(questionsColumns...).
-		From("questions").
-		ToSql()
-
-	rows, err := query(ctx, sql, boundArgs...)
+		From("questions")
 
-	if err != nil {
-		log.Println(err)
-		return nil, err
-	}
-
-	return pgx.CollectRows(rows, func(row pgx.CollectableRow) (*Question, error) {
-		q := Question{}
-		err := row.Scan(&q.QuestionID, &q.Question, &q.NotesPrompt, &q.Order, &q.PillarID)
-		return &q, trapError(err)
-	})
+	return query(ctx, sqlb, pgx.RowToAddrOfStructByNameLax[Question])
 }
 
 func FindQuestionByID(ctx context.Context, questionID int32) (*Question, error) {
-	sql, boundArgs, _ := sqlBuilder.
+	sqlb := stmntBuilder.
 		Select(questionsColumns...).
 		From("questions").
-		Where("questionid=?", questionID).
-		ToSql()
+		Where("questionid=?", questionID)
 
-	row, err := queryRow(ctx, sql, boundArgs...)
-
-	if err != nil {
-		log.Println(err)
-		return nil, err
-	}
-
-	q := Question{}
-	err = row.Scan(&q.QuestionID, &q.Question, &q.NotesPrompt, &q.Order, &q.PillarID)
-	return &q, err
+	return queryRow(ctx, sqlb, pgx.RowToStructByNameLax[Question])
 }
 
 // FindQuestionsByFismaSystem joins questions with functions to return questions relevant to the fismasystem as determined by the datacenterenvironment.
 // It is used by all users to list questions relevant to the specified fisma system
 func FindQuestionsByFismaSystem(ctx context.Context, fismaSystemID int32) ([]*Question, error) {
-	sql, boundArgs, _ := sqlBuilder.
+	sqlb := stmntBuilder.
 		Select("questions.questionid, question, notesprompt, questions.ordr, pillars.pillarid, pillars.pillar, pillars.ordr, functionid, function, description").
 		From("questions").
 		InnerJoin("pillars ON pillars.pillarid=questions.pillarid").
 		InnerJoin("functions ON functions.questionid=questions.questionid").
 		InnerJoin("fismasystems ON fismasystems.datacenterenvironment=functions.datacenterenvironment AND fismasystems.fismasystemid=?", fismaSystemID).
-		OrderBy("pillars.ordr, questions.ordr ASC").
-		ToSql()
-
-	rows, err := query(ctx, sql, boundArgs...)
-
-	if err != nil {
-		log.Println(err)
-		return nil, err
-	}
+		OrderBy("pillars.ordr, questions.ordr ASC")
 
-	return pgx.CollectRows(rows, func(row pgx.CollectableRow) (*Question, error) {
+	return query(ctx, sqlb, func(row pgx.CollectableRow) (*Question, error) {
```

Comment:
One of the rare examples of where a custom function is passed to account for result fields that don't align to the struct exactly